src/components/App/App.jsx

```diff
@@ -0,0 +1,116 @@
+import Navbar from "../Navbar/Navbar"
+import Home from "../Home/Home"
+import Profile from "../Profile/Profile"
+import Recipes from "../Recipes/Recipes"
+import Pantry from "../Pantry/Pantry"
+import ShoppingCart from "../ShoppingCart/ShoppingCart"
+import './App.css';
+
+import { BrowserRouter, Routes, Route } from "react-router-dom"
+import { useState, useEffect } from "react"
+
+export default function App() {
+
+  var basicProducts = [
+    {name: "apple",
+    quantity: 2},
+    {name: "orange",
+    quantity: 5},
+    {name: "grape",
+    quantity: 8},
+    {name: "bacon",
+    quantity: 3},
+    {name: "milk",
+    quantity: 2},
+    {name: "pasta",
+    quantity: 1}
+  ]
+
+  var basicProductForm = {
+    name: "",
+    quantity: ""
+  }
+
+const [products, setProducts] = useState(basicProducts)
+const [productForm, setProductForm] = useState(basicProductForm)
+
+const handleProductQuantity = (productName, operation) => {
+  // find index of productName
+  let itemIndex = products.findIndex(item => item.name === productName)
+
+  let newProducts = []
+  for (let i = 0; i < products.length; i++) {
+    newProducts.push(products[i])
+  }
+
+  if (operation == "add") {
+    newProducts[itemIndex].quantity += 1
+  } 
+  else if (operation === "minus") {
+    newProducts[itemIndex].quantity -= 1
+    if (newProducts[itemIndex].quantity == 0) {
+      newProducts.splice(itemIndex, 1)
+    }
+  } else {
+    newProducts.splice(itemIndex, 1)
+  }
+  
+  setProducts(newProducts)
+}
+
+const handleOnSubmitProductForm = () => {
+  let itemName = productForm.name.toLowerCase()
+  // find index of productName
+  let itemIndex = products.findIndex(item => item.name === itemName)
+
+  let newProducts = []
+  for (let i = 0; i < products.length; i++) {
+    newProducts.push(products[i])
+  }
+
+  if (itemIndex === -1) {
+    let newItem = {
+      name: itemName,
+      quantity: Number(productForm.quantity)
+    }
+    newProducts.push(newItem)
+  } else {
+    newProducts[itemIndex].quantity += Number(productForm.quantity)
+  }
+
+  setProducts(newProducts)
+  setProductForm(basicProductForm)
+}
+
+const handleOnProductFormChange = (event) => {
+  let key = event.target.name
+  let val = event.target.value
+  let newProductForm = {
+    name: productForm.name,
+    quantity: productForm.quantity
+  }
+
+  newProductForm[key] = val
+  
+  setProductForm(newProductForm)
+}
+
+  return (
+    <div className="App">
+      <BrowserRouter>
+        <main>
+          <Navbar />
+          <Routes>
+            <Route path="/" element={<Home />}/>
+            <Route path="/profile" element={<Profile />}/>
+            <Route path="/recipes" element={<Recipes />}/>
+            <Route path="/pantry" element={<Pantry productForm={productForm} handleOnProductFormChange={handleOnProductFormChange} handleOnSubmitProductForm={handleOnSubmitProductForm} handleProductQuantity={handleProductQuantity} products={products}/>}/>
```

Comment:
Question: I'm not super familiar with typical React structure, but should some of the productForm logic live inside the Pantry component instead of in App.jsx? It seems like App.jsx will quickly grow out of hand if all of this logic needs to be added here? 

Feel free to wait until later to move this if that's what you want to do.

Reply:
I am not sure, I will look into this because I assumed convention was to include all the handle functions in App.jsx so we could pass the data to multiple components. 
I'll move this into separate components in a later PR once I determine that the other components won't need the same information (I will add a TODO to remember this)